dandiapi/api/views/zarr.py

```diff
@@ -94,6 +99,28 @@ class ZarrViewSet(ReadOnlyModelViewSet):
     lookup_field = 'zarr_id'
     lookup_value_regex = ZarrArchive.UUID_REGEX
 
+    def get_queryset(self):
+        query_serializer = ZarrListSerializer(data=self.request.query_params)
+        query_serializer.is_valid(raise_exception=True)
+
+        # Filter by search params
+        queryset = self.queryset
+        data = query_serializer.validated_data
+        if 'dandiset' in data:
+            queryset = queryset.filter(dandiset=int(data['dandiset'].strip('0')))
+        if 'name' in data:
+            queryset = queryset.filter(name=data['name'])
+
+        return queryset
+
+    @swagger_auto_schema(
+        query_serializer=ZarrListSerializer(),
```

Comment:
I'm not sure what you mean? What is being limited here?

Reply:
I misread this, I thought it was defining `ZarrListSerializer` as the response serializer for the list endpoint. It's almost a plausible thing to do, since you might not want too much information included when you are listing lots of zarrs, but it seemed like a strange thing to add here.